lib/hydra/works/models/concerns/generic_file_behavior.rb

```diff
@@ -20,6 +20,42 @@ module GenericFileBehavior
       include Hydra::Works::GenericFile::Derivatives
       include Hydra::Works::GenericFile::MimeTypes
       include Hydra::Works::GenericFile::VersionedContent
+      include Hydra::Works::BlockChildObjects
+
+      filters_association :members, as: :child_generic_files, condition: :works_generic_file?
+    end
+
+    module ClassMethods
+      def indexer
+        Hydra::Works::GenericFileIndexer
+      end
+    end
+
+    # @return [Boolean] whether this instance is a Hydra::Works Collection.
+    def works_collection?
+      false
+    end
+
+    # @return [Boolean] whether this instance is a Hydra::Works Generic Work.
+    def works_generic_work?
+      false
+    end
+
+    # @return [Boolean] whether this instance is a Hydra::Works Generic File.
+    def works_generic_file?
+      true
+    end
+
+    def parents
+      aggregated_by
+    end
+
+    def parent_generic_works
+      aggregated_by.select { |parent| parent.class.included_modules.include?(Hydra::Works::GenericWorkBehavior) }
```

Comment:
I can see your point that it is new to Hydra:Works.  If you want to go the route of doing this different from what is at PCDM, then I will extract out all the parent code in Hydra:Works and open a separate issue fixing pcdm parents and adding parents in Hydra:Works.  I believe it can be removed without breaking other parts.